### DISCUSSION TITLE
Can you add RLS policies to storage buckets on local dev?

### DISCUSSION THREAD

**Discussion by thibistaken** [supabase/supabase#18993] [Questions] [answered]
I am running Supabase in local development, as well as staging and production environments. In the local dev env, I've added RLS policies to INSERT, SELECT and DELETE files in a storage bucket but when I run `supabase db diff`, I get: `No schema changes found`. I've had correct SQL statements that were generated on the Supabase Studio UI, but it seems they are not detected for the migrations. I've ran other successful `supabase db diff` commands prior, so I know this works on the public schema for instance. 

I wanted to take the generated SQL statements for the storage bucket, and add them manually in migrations I would then run but in Storage > Configuration > Policies, when I click edit > review, I get this message: `There are no changes made to this policy` so I cannot even view the previously generated SQL statements for the RLS policies. 

Any help? Thanks!

**Comment by ilbertt** [comment, marked as answer]
Have you tried specifying the schemas that you want to diff?
```bash
supabase db diff --schema public,storage
```

Same for any other schema that you want to include in the migrations, like the `extensions` schema:
```bash
supabase db diff --schema public,storage,extensions
```

**Reply by thibistaken** [reply]
Thank you. That's it! I had not tried, but this works. I just did not see it in the docs at first... but it is there: https://supabase.com/docs/reference/cli/supabase-db-diff?example=supabase-db-diff-specific-schema

**Reply by thibistaken** [reply]
Btw I had created a similar issue on the main Supabase repo, for which I had a similar answer. Perhaps this topic is better suited for discussions here so I will suggest to close the issue: https://github.com/supabase/supabase/issues/18996
